pkg/containers/containers.go

```diff
@@ -169,58 +169,54 @@ func (c *Containers) cgroupLookupUpdate(cgroupId uint64) error {
 // add cgroupId with a matching container id, extracted from path
 func (c *Containers) CgroupUpdate(cgroupId uint64, path string) (CgroupInfo, error) {
 	info := CgroupInfo{Path: path}
+	info.ContainerId, info.Runtime = getContainerIdFromCgroup(path)
 
-	for _, pc := range strings.Split(path, "/") {
-		if len(pc) < 64 {
-			continue
-		}
-		containerId, runtime := getContainerIdFromCgroup(pc)
-		if containerId == "" {
-			continue
-		}
-		info.ContainerId = containerId
-		info.Runtime = runtime
-		// we want to get container info with respect to host
-		break
-	}
 	c.mtx.Lock()
 	c.cgroups[uint32(cgroupId)] = info
 	c.mtx.Unlock()
 	return info, nil
 }
 
 // extract container id and container runtime from path
-func getContainerIdFromCgroup(pathComponent string) (string, string) {
-	runtime := "unknown"
-	path := strings.TrimSuffix(pathComponent, ".scope")
-
-	if strings.HasPrefix(path, "docker-") {
-		runtime = "docker"
-		path = strings.TrimPrefix(path, "docker-")
-		goto check
-	}
-	if strings.HasPrefix(path, "crio-") {
-		runtime = "crio"
-		path = strings.TrimPrefix(path, "crio-")
-		goto check
-	}
-	if strings.HasPrefix(path, "cri-containerd-") {
-		runtime = "containerd"
-		path = strings.TrimPrefix(path, "cri-containerd-")
-		goto check
-	}
-	if strings.HasPrefix(path, "libpod-") {
-		runtime = "podman"
-		path = strings.TrimPrefix(path, "libpod-")
-		goto check
-	}
+func getContainerIdFromCgroup(cgroupPath string) (string, string) {
+	prevPathComp := ""
+	for _, pc := range strings.Split(cgroupPath, "/") {
+		if len(pc) < 64 {
+			prevPathComp = pc
+			continue
+		}
 
-check:
-	if matched, _ := regexp.MatchString(`^[A-Fa-f0-9]{64}$`, path); !matched {
-		return "", ""
+		runtime := "unknown"
+		id := strings.TrimSuffix(pc, ".scope")
+
+		switch {
+		case strings.HasPrefix(id, "docker-"):
+			runtime = "docker"
+			id = strings.TrimPrefix(id, "docker-")
+		case strings.HasPrefix(id, "crio-"):
+			runtime = "crio"
+			id = strings.TrimPrefix(id, "crio-")
+		case strings.HasPrefix(id, "cri-containerd-"):
+			runtime = "containerd"
+			id = strings.TrimPrefix(id, "cri-containerd-")
+		case strings.HasPrefix(id, "libpod-"):
+			runtime = "podman"
+			id = strings.TrimPrefix(id, "libpod-")
+		}
+
+		if matched, _ := regexp.MatchString(`^[A-Fa-f0-9]{64}$`, id); matched {
+			if runtime == "unknown" && prevPathComp == "docker" {
```

Comment:
can we have a case with non systemd containerd or podman? meaning for example a path like `				/containerd/01adbaf26db7fac62f8b7ec5b116b2e60f4abbf230e29122e667996c6b5c04d4/`?
in that case, wouldn't we still have runtime `unknown`?

Reply:
I am not aware of the existance of such cgroups paths for containerd/podman.
Are you?
Anyways, we can easilly add those in the future if it will be required